File: unix/learning-c/playground.js

const obj = { name: 'faheem' };

let n = 6;


function modify(o, n) {
    o.name = 'Faheem Khan';
    n = 6000;
}

console.log(obj)
console.log(n);

modify(obj, n);

console.log(obj)
console.log(n);

File: unix/learning-c/playground.c
#include <stdio.h>

struct Object {
  const char *name;
};

void modify(struct Object *o, int n) {
  (*o).name = "Dylan";
  n = 1000;
}

int main() {
  struct Object obj;
  obj.name = "Joe";
  int num = 700;

  printf("Before modifications:\n");
  printf("obj.name: %s\n", obj.name);
  printf("num: %d\n", num);

  modify(&obj, num);

  printf("After modifications:\n");
  printf("obj.name: %s\n", obj.name);
  printf("num: %d\n", num);

  //   printf(obj);

  return 0;
}